test/base.test.js

```diff
@@ -669,3 +669,33 @@ test('skip the cache if operation is Mutation', async ({ equal, same, teardown }
   equal(skipCount, 0)
   equal(hitCount, 0)
 })
+
+test('using all option as string', async (t) => {
+  const app = fastify()
+  app.register(mercurius)
+  app.register(cache, {
+    all: 'true'
+  })
```

Comment:
besides checking that this throws the right error can you check if awaiting the registration would work too? instead of having to await app.ready()? E.g.

```suggestion
  await t.rejects(app.register(cache, {
    all: 'true'
  })) // TODO: check also that the error is the one you expect
  
  // then remove app.ready()
```